pkg/controller/reconciler/reconciler.go

```diff
@@ -74,31 +70,31 @@ func NewComponentReconciler(
 	return &cc, nil
 }
 
-func (r *ComponentReconciler) GetRelease() cluster.Release {
+func (r *Reconciler[T]) GetRelease() cluster.Release {
 	return r.Release
 }
 
-func (r *ComponentReconciler) GetLogger() logr.Logger {
+func (r *Reconciler[T]) GetLogger() logr.Logger {
 	return r.Log
 }
 
-func (r *ComponentReconciler) AddOwnedType(gvk schema.GroupVersionKind) {
+func (r *Reconciler[T]) AddOwnedType(gvk schema.GroupVersionKind) {
 	r.m.AddGVK(gvk, true)
 }
 
-func (r *ComponentReconciler) Owns(obj client.Object) bool {
+func (r *Reconciler[T]) Owns(obj client.Object) bool {
 	return r.m.Owns(obj.GetObjectKind().GroupVersionKind())
 }
 
-func (r *ComponentReconciler) AddAction(action actions.Fn) {
+func (r *Reconciler[T]) AddAction(action actions.Fn) {
 	r.Actions = append(r.Actions, action)
 }
 
-func (r *ComponentReconciler) AddFinalizer(action actions.Fn) {
+func (r *Reconciler[T]) AddFinalizer(action actions.Fn) {
 	r.Finalizer = append(r.Finalizer, action)
 }
 
-func (r *ComponentReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+func (r *Reconciler[T]) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
```

Comment:
so i guess the removal line 114+ is because now this logic is for both component and service. 
do you feel (not in the current code) one day we will still need a `rr.DSC.XXXX` for either component or service?

Reply:
For component is should definitively not needed as the components are created out of the DSC, so they should have everything they needs as part of the spec, if not, we should probably revisit the specific component.  

As long term, the internal APIs should avoid using DSCI and DSC so i.e. if a component requires any authentication information, it should use the **status** of the related auth internal API. Yeah, currently not possible but that should be the ultimate goal so the less we depends on them, the better.